planner/mapper/mapper.go

```diff
@@ -797,50 +803,66 @@ func resolveInnerFilterDependencies(
 	source map[string]any,
 	mapping *core.DocumentMapping,
 	existingFields []Requestable,
+	resolvedFields []Requestable,
 ) ([]Requestable, error) {
 	newFields := []Requestable{}
 
-sourceLoop:
 	for key := range source {
-		if strings.HasPrefix(key, "_") && key != request.KeyFieldName {
+		if key == opAnd || key == opOr { // handle _not
```

Comment:
todo: I don't think the `// handle _not` comment belongs in develop branch, this is basically a todo, and if presered it should be linked to a ticket.

Reply:
it's something I'm working on now. Will push a bit later to this PR